utils/problems.py

```diff
@@ -47,12 +47,25 @@ class Submissions:
     hard: int
     score: int
 
+@dataclass 
+class LanguageProblemCount:
+    language : str 
+    problem_count: int 
+
+@dataclass 
+class SkillProblemCount:
+    skill : str 
+    problem_count: int 
+
 
 @dataclass
 class UserStats:
     real_name: str
     submissions: Submissions
-
+    languages_problem_count: list #Made up of LanguageProblemCount objects
+    fundamental_skills_problem_count: list #Made up of SkillProblemCount objects
+    intermediate_skills_problem_count: list #Made up of SkillProblemCount objects
+    advanced_skills_problem_count: list #Made up of SkillProblemCount objects
```

Comment:
rename the to match my suggestions from below, e.g. `tag_problem_counts_advanced`. We could start the variable name with the difference (aka the difficulty such as advanced, intermediate, fund...), but as per my suggestion, I think in this case we should start the class attribute names with the common part of the name. I hope that makes sense, pretty waffly explanation from me so mb

Reply:
No need to add those comments. We can write self-documenting code, by actually type hint what the list is made up of, which is cool! So rather than type hint just `: list`, say `: list[LanguageProblemCount]`. Just for your knowledge, type hints aren't actually ever utilised by python during compilation, they're merely there to help programmers as they provide readable insights to the variable's type, but also aid in your IDE's intellisense.